src/lib/data.ts

```diff
@@ -1,13 +1,17 @@
 import fs from 'fs'
+import { createDomainCache, createIdCache } from '../helpers/cache'
 import { DB, ServiceKey } from '../types/admin'
-import { Mapping, Domain } from '../types/general'
+import { Mapping, MappingObj, Domain } from '../types/general'
 
 const data: DB = {
   serviceKeys: [],
   mappings: [],
   availableDomains: []
 }
 
+let mappingsCache: MappingObj | {} = {}
+let mappingsDict: MappingObj | {} = {}
+
```

Comment:
Below functions, you can just call this function instead of always writing the `if` statement: `updateCache(table)`
```
const updateCache = (table) => {
  if (table === 'mappings') {
    mappingsCache = createDomainCache(data.mappings)
    mappingsDict = createIdCache(data.mappings)
  }
}
```